### PR TITLE
refactor: improve logging for hostRules

### PR DESCRIPTION
**Pull request by rarkins**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/http/host-rules.ts [hunk 1]
```diff
@@ -120,24 +120,30 @@ export function applyHostRules<GotOptions extends HostRulesGotOptions>(
   const options: GotOptions = { ...inOptions };
   const foundRules = findMatchingRules(options, url);
   const { username, password, token, enabled, authType } = foundRules;
+  const { host } = new URL(url);
```
Comment (viceice) [inline lib/util/http/host-rules.ts:123]: this throws for invalid url.

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/util/http/host-rules.ts (hunk 1 above)
Comment (rarkins) [reply lib/util/http/host-rules.ts:123]: I'm not sure it can be invalid by this point - will check

Comment (rarkins) [reply lib/util/http/host-rules.ts]: Added try/catch

File: lib/util/http/host-rules.ts [hunk 2]
```diff
@@ -120,24 +120,35 @@ export function applyHostRules<GotOptions extends HostRulesGotOptions>(
   const options: GotOptions = { ...inOptions };
   const foundRules = findMatchingRules(options, url);
   const { username, password, token, enabled, authType } = foundRules;
+  let host: string | undefined;
+  try {
+    host = new URL(url).host;
```
Comment (viceice) [inline lib/util/http/host-rules.ts:125]: why not using parseUrl?

https://github.com/renovatebot/renovate/blob/13960240da08a4c85faa5ec898a8b7fb0b9ad648/lib/util/url.ts#L103

**Review by viceice — APPROVED** [review]

**Pull request merged**